Another connect without context

nyalldawson opened the pull request.

nyalldawson committed:
Another connect without context

nirvn commented on src/gui/providers/ogr/qgsgeopackageitemguiprovider.cpp:
```diff
@@ -94,7 +94,7 @@ void QgsGeoPackageItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu
     QAction *actionAddTable = new QAction( tr( "Create a New Layer or Table…" ), collectionItem->parent() );
     QPointer<QgsGeoPackageCollectionItem>collectionItemPtr { collectionItem };
-    connect( actionAddTable, &QAction::triggered, [ collectionItemPtr ]
+    connect( actionAddTable, &QAction::triggered, actionAddTable, [ collectionItemPtr ]
```
Out of curiosity, why not using the collectionItemPtr pointer instead of the actionAddTAble pointer here? In theory, isn't a signal sent to a null receiver harmless (in that it'll merely not send the signal)?

nyalldawson replied:
I've pushed an improved version now!

nyalldawson committed:
Improved lambda capture

Even if the collection item is deleted before the action is triggered,
we can still safely show the creation dialog and act on it (we just
can't trigger the item refresh automatically)

nirvn commented:
@nyalldawson , improved it is indeed.
